components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag

```diff
@@ -0,0 +1,429 @@
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~      http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+-->*/%>
+<%
+var update = request.getParameter("update");
+var headers = {"Authorization":"Basic " + auth};
+var start = 0;
+
+
+var processNames = get( httpUrl + "/bpmn/repository/process-definitions?size=0" ,null, headers, "json" ).data;
+var processBarChart = [];
+for(var i= 0; i < processNames.data.length; i++){
+	var name = processNames.data[i].id;
+	var processCount = get( httpUrl + "/bpmn/history/historic-process-instances?processDefinitionId=" + name ,null, headers, "json" ).data.total;
+	processBarChart.push([
+		name, processCount
+	]);
+}
+
+var minProcessCount = [];
+var maxProcessCount = [];
+
+  if(update != null && update === "true"){
+    if(option === "processinstancecount"){
+
+    var pInstanceCount = request.getParameter("id");
+    if(pInstanceCount == "min") {
+      sortAscending(processBarChart);
+      minProcessCount = processBarChart.slice(0,10);
+      processBarChart = minProcessCount;
+    }else if(pInstanceCount =="max"){
+      sortDescending(processBarChart);
+      maxProcessCount = processBarChart.slice(0,10);
+      processBarChart = maxProcessCount;
+    }
+    else if(pInstanceCount=="all"){
+      processBarChart=processBarChart;
+    }
+  }
+}
+
+var listProcess = get( httpUrl + "/bpmn/history/historic-process-instances?finished=true&size=0" ,null, headers, "json" );
+var distListProcess = get( httpUrl + "/bpmn/repository/process-definitions?finished=true&size=0" ,null, headers, "json" );
+
+var processBarChartAvgTime = [];
+var taskArray = {};
+
+for (var i in listProcess.data.data){
+   var totalTime = 0;
+   var processName = listProcess.data.data[i].processDefinitionId;
+   var count = get( httpUrl + "/bpmn/history/historic-process-instances?finished=true&processDefinitionId=" + processName ,null, headers, "json" ).data.total;
+
+   var timeduration = get( httpUrl + "/bpmn/history/historic-process-instances?finished=true&processDefinitionId=" + processName ,null, headers, "json" );
+
+      for(var i in timeduration.data.data){
+        var timeForOne = timeduration.data.data[i].durationInMillis;
+        var timeInMinutes = ((timeForOne /(1000*60))%60);
+        totalTime = totalTime + timeInMinutes;
+      }
+
+   var averageTime = totalTime / count;
+   var avgRoundedTime = Math.round(averageTime * 1000) / 1000;
+
+    if(checkForDuplicates(processBarChartAvgTime,processName) == 1){
+
+    }else{
+
```

Comment:
The unnecessary spaces has been removed when refactoring the code , so this issue is already fixed.
https://github.com/wso2/carbon-business-process/blob/master/components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag